Fix: #1370 Document environment for running in production mode

SuYaSh-PaThAk04 opened the pull request:
**Summary**:
<!--
What existing issue does the pull request solve?
Please provide enough information so that others can review your pull request
-->This pull request updates the production environment documentation as requested in issue #1370.

If this closes an existing issue then add "closes #xxxx", where xxxx is the issue number

**Description for the changelog**:
<!-- A short (one line) summary that describes the changes in this pull request for inclusion in the change log -->
Documented the required environment variables for running Threat Dragon in production mode.
**Declaration**:
This pull request includes text generated with the assistance of a generative AI tool. All AI-generated content has been reviewed and edited for accuracy.
- [x] appropriate unit tests have been created / modified
- [x] functional tests created / modified for changes in functionality
- [x] any use of AI has been declared in this pull request

**Other info**:
<!-- Add here any other information that may be of help to the reviewer -->
This is how it looks
<img width="1335" height="719" alt="Screenshot 2025-11-15 113909" src="https://github.com/user-attachments/assets/017c8ce4-2e98-4ea6-8efb-902821123f67" />


Thanks for submitting a pull request!
Please make sure you follow our [Code of Conduct](../code_of_conduct.md)
and our [contributing guidelines](../contributing.md)

jgadsden reviewed (changes requested):
This is not what is needed for the documentation - it should not be as block copy of the issue description, but instead it should document what is often (but not always) needed for a production environment with discussion on only the environment variables that probably need to change

SuYaSh-PaThAk04 commented:
> @SuYaSh-PaThAk04 the documentation for Threat Dragon has to be accurate, are you using generative AI for this PR? If you are then it has to be declared, hence the tick box where you agree to that
> 
> We do not use env variables such as CORS_ORIGIN, and the original issue was specific to the environment variables specific to Threat Dragon

Hi @jgadsden — thanks for the feedback and for flagging the scope.

I did use a generative AI tool to help draft the initial wording and I’ve declared that in the PR (per the project rules). I’ve critically reviewed and edited the content afterwards to ensure accuracy and project-specific wording.

I removed the block-copy of the issue and replaced it with documentation that focuses only on environment variables typically changed in production, with short explanations and guidance for each. I avoided recommending variables that Threat Dragon does not use by default.

If there are specific Threat Dragon environment variables I missed, or any variables that are incorrect for the project, please point them out and I’ll update immediately.
